### DISCUSSION TITLE
🔥 ✨ Request for Feedback: Organization Discussions

### DISCUSSION THREAD

**Discussion by jayceeday** [community/community#17669] [Discussions]
I'm Jaycee, a Designer on the Discussions team at GitHub!

We launched [Organization Discussions](https://github.blog/changelog/2022-04-12-organization-discussions/) a month ago and are interested in what you think! 

What do you like? What do you not like? If you could wave a magic wand what would you ask for? 🪄

**Comment by MrKrishnaAgarwal** [comment]
Great
Keep working on it 
Have a great day

**Reply by jayceeday** [reply]
Thank you @MrKrishnaAgarwal 🤗

Was there anything in particular you like about it?

**Reply by MrKrishnaAgarwal** [reply]
Yes,
It is a great way to connect with people.
I find GitHub is becoming innovative day by day.
Great work done by you and the whole @github-staff.

**Reply by divyaGrvgithub** [reply]
I'm Jaycee, a Designer on the Discussions team at GitHub!

We launched [Organization Discussions](https://github.blog/changelog/2022-04-12-organization-discussions/) a month ago and are interested in what you think!

What do you like? What do you not like? If you could wave a magic wand what would you ask for? 🪄

**Reply by AmirrezaNasrollahzade** [reply]
https://github.com/orgs/community/discussions/categories/discussions

**Comment by jimmo** [comment]
Thanks @jayceeday, we love this feature!

Just wondering if it would be possible to remove the "This is a ✨special✨ repository containing the organization level discussions for micropython. Everything posted here will also be visible at the organization level." message.

We have several repos under the main org, but 99% of issues and PRs go to the "primary" repo, and most of the people using our discussions get there from the Discussions tab on that repo. (We deliberately chose to use our primary repo for discussions for this reason, so that the tab would be there, also because it wasn't clear how transferring issues to discussions would work otherwise).

So ideally, clicking on the "Discussions" tab from the primary repo would just take them to the organisation discussions. Instead they see this view and the "special repository" banner message is a bit confusing in an "am I in the right place?" sort of way, as well as taking up about 15% of the vertical screen real estate on my laptop.

Thanks!

**Reply by jayceeday** [reply]
Hey @jimmo 

Sorry for a delayed response as I was on vacation. 

This is really great feedback and I appreciate the detail. I will look into this, thank you! 🙏

**Reply by jsoref** [reply]
@jayceeday: Did you get an answer?

**Comment by ssbarnea** [comment]
I really appreciate that you enabled organization level discussion as this is addressing (half) of the big scalability issue for organizations with many related repos.

The one forum for each repo approach does only work when each repository is a clearly defined project that does not have things in common with other ones. As we can guess, stuff under a single organization can often be related. A real project could easily span 2-3 repos or even 10 and nobody would be able to watch and moderate 10 different threads, especially as people posting there might post to the wrong place very often or fail to search for existing question in the right place.

Still, there is a **critical missing feature**, one that makes unsuitable to use organization discussions. The fact that you do not get a "discussion" tab on each repository makes the discoverability of the organization discussion very hard for users. Most of them would assume that you did never enable the discussions on that repo.

Fixing this should not be too hard, but until you do, I do not see much practical use on enabling them, as they would be hard to find.

Ideally the link to discussions should link to a specific category in organization discussions, but I am sure that this might be harder to implement, so even a link to root of org discussions could be seen as acceptable.

I hope this helps and I would appreciate any insights regarding on how you plan to address it, even if I know that it would not be possible to get any assurances regarding when these features would ship.

**Reply by amacneil** [reply]
Agree with this. We would like to have the "Discussion" tab on several of our repos link to the org-level discussions, to make it easier for people to find.

**Reply by jimmo** [reply]
We'd love to see this too. Although we have one "main" repo, there are several other repos in our org that share the same discussion forum, but don't get the "Discussions" tab.

**Reply by amacneil** [reply]
Also I would really like the ability to directly migrate issues from a repo to the org-wide discussion board. Currently this requires a two step process of moving the issue to our `community` repo, then converting it to a discussion, which leaves a rather messy trail. And if you do it in the other order (convert to discussion, _then_ migrate to the `community` repo) it breaks the link entirely giving users no indication why their issue was closed.

**Reply by jayceeday** [reply]
Thank you for the feedback this is great, and something we will consider!

To clarify, is the problem that you aren't seeing discussions from other repos within the org on the org discussion? 
Or, would you prefer a link to have people go to the Org Discussions to post instead of in the repo? 
Or something else?

Thanks!

**Reply by amacneil** [reply]
Organization discussions as they exist today are very frustrating. For example:

We have an open source project https://github.com/foxglove/studio and another one https://github.com/foxglove/mcap. And we have org discussions at https://github.com/orgs/foxglove/discussions. Some of our discussion categories map directly to open source repos, and others relate to closed source projects.

These are all detached from each other. There is no way to direct users from the individual repos to the correct category in org discussions. There is no way to convert an issue in a repo to a discussion at the org level. 

I can see a few ways to resolve this:
1. Repo-level discussions could be removed entirely, and all discussions could exist at the org level. Each repo could configure which "category" in the org discussions they want to link to when you click the "Discussions" tab on that repo.
2. Org-level discussions could be removed entirely, and replaced with basically a summary view of all discussions from all repos in that org. If you do this, it needs to still be easy to make new discussions for both open source and closed source topics.
3. Discussions could stay the way they are today (both org and repo-level discussions), but for each repo I have the option to redirect the repo Discussions tab over to a specific category of my org-discussions. When this is enabled, the "Convert to discussion" button on issues in that repo would also automatically get migrated over to the org-level discussions.

**Reply by lars-reimann** [reply]
> To clarify, is the problem that you aren't seeing discussions from other repos within the org on the org discussion? Or, would you prefer a link to have people go to the Org Discussions to post instead of in the repo? Or something else?
> 
> Thanks!

In my case the issue is that people visit a repository and don't see the discussions tab. So they don't know where to ask questions. At best they create an issue, at worst they move somewhere else. So I'd like the organization discussions to still be visible in some of the org's repositories.

When discussions are enabled in a repository, a user could, for example, choose with radio buttons (or a dropdown) where the discussions should be located:

![image](https://user-images.githubusercontent.com/2501322/226301677-b14f531e-1063-4ef8-80bf-e5b11c9dc9b4.png)

Why not use repository discussions then? Questions might be related to or relevant for multiple repositories, so I'd rather have them all in one place and organize them with categories etc.

**Edit:** I've now linked to the discussions using a `contact_link` in `ISSUE_TEMPLATE/config.yml`. Now people at least learn about the forum when they try to create a new issue. I'd still prefer if the tab "Discussions" would be always visible in the repo as described above.

**Reply by jsoref** [reply]
@jayceeday: @lars-reimann's [recap](https://github.com/orgs/community/discussions/17669#discussioncomment-5366028) is a perfect description of the needs described in this thread. For repositories that don't themselves host discussions, but for which there are discussions, users should be able to easily find them without anyone having to bend over backward (e.g. editing the `ISSUE_TEMPLATE/config.yml` to link to the org discussions).

**Comment by dudkinox** [comment]
Hi Jaycee,

Thanks for asking for my feedback on Organization Discussions! I haven't had the chance to use it yet, so I don't have any specific thoughts on it. However, as a general suggestion, I think it would be helpful to have some sort of notification system in place so that users can stay updated on new discussions or updates to existing discussions within an organization. This could be in the form of email notifications, or perhaps a notification feed within the app itself.

I also think it would be useful to have some sort of organization hierarchy or categorization system in place, so that users can easily find and participate in discussions that are relevant to their role or interests within the organization.

Overall, I think it's great that you're looking for feedback on Organization Discussions, and I hope these suggestions are helpful!

**Reply by jimmo** [reply]
@dudkinox When you add Discussions to your repo, then they notify just like Issues and PRs do (both by email and the app notification feed).

One frustrating thing is that the emails don't have any distinguishing headers to tell them apart from issues and PRs. (See https://github.com/orgs/community/discussions/59#discussioncomment-3510061 )

And for hierarchy/categorisation, see https://github.com/orgs/community/discussions/32323  (we would really like this too for our org. It's enabled on the GitHub Community Discussions, just not on any others). And for notifying on individual categories see https://github.com/orgs/community/discussions/3951#discussioncomment-3879939

**Reply by amacneil** [reply]
It would be nice to subscribe to specific categories, rather than all discussions in the org.

**Reply by jayceeday** [reply]
Thank you for the feedback, this is great feedback and some things we will consider!

@dudkinox would love for you to try it out, and then let us know what you think

**Comment by amacneil** [comment]
How can we nested discussion categories, the way GitHub has done at https://github.com/orgs/community/discussions?

**Reply by jayceeday** [reply]
This is only available in the GitHub community feedback, but something we are already considering to offer as a feature. You can vote/leave comments[ in this Discussion](https://github.com/orgs/community/discussions/32323)

**Comment by vico1973** [comment]
`[Sexo ](Fuckbook.tv ) `

**Comment by ntessore** [comment]
I think organisation discussions are great. Slight nitpick: discussions appearing on the organisation page under "Top discussions this past month" link to the discussion in its repository, not the organisation.

**Reply by jayceeday** [reply]
Thank you, this is a bug so I have reported it!

**Comment by panditakshay402** [comment]
Its a great initiate, This will  help users to get the  answer  easily.

**Comment by AlbertThor** [comment]
Is it possible to have a organization discussions on a repo that has been forked to the organization? I don't see the forked repo in the transfer-discussion tab when I try this...

**Reply by jsoref** [reply]
Yes, it's possible, but you have to bend over backwards.
See https://github.com/orgs/check-spelling/discussions/1 for an example.

Steps:
1. Create a [discardable repository in the source organization](https://github.com/check-spelling/migrating-repository/)
2. Enable discussions
3. Transfer the discussion from the source repository to the [discardable repository](https://github.com/check-spelling/migrating-repository/discussions/1)
4. Transfer the discardable repository to the [destination organization](https://github.com/check-spelling-sandbox/migrating-repository)
5. Transfer the discussion to the [organization discussions repository](https://github.com/orgs/check-spelling-sandbox/discussions/1)
6. (Not implemented for this demonstration: do something to the migrating repository -- you could just keep ping-ponging it back and forth as you need to shuttle discussions around.)

Obviously to do this, you need to be either and admin on both sides, or have access to someone who can accept the transfers. In my case, I had to use Passkey to authorize the repository to move.

**Comment by AlexdeSouza2502** [comment]
como faço para melhorar meu desempenho nos estudos de codigo

**Comment by sacramentosolutions** [comment]
I appreciate that you enabled organization level discussion!

**Comment by 3103903362** [comment]
因为其他事务的原因，我有一段时间没有使用了，再次进入发现页面发生了一些变化，但是感觉新页面没以前的页面方便，而且每次进入网站但会有严重的卡顿，有时甚至无法进入，这个问题能否解决呢？

**Comment by tharodo** [comment]
The launch of Organization Discussions on GitHub is a fantastic step forward. It's awesome to have a dedicated space for structured conversations.

I'm really liking the improved collaboration and focused discussions. It's a boost to teamwork and project development. One thing I'd love to see is more customization options for the discussion boards, allowing organizations to align the platform with their unique style.

If I could wave that magic wand, I'd wish for AI-powered sentiment analysis integration. This could help monitor discussions' health and integration with project management tools for a smoother transition from talk to action.

Thanks for reaching out and looking forward to seeing how Organization Discussions evolves!

**Reply by AmirrezaNasrollahzade** [reply]
your welcome

**Comment by 3103903362** [comment]
我登录不上去了，所以无法再次评论



---Original---
From: "King ***@***.***&gt;
Date: Sun, Aug 20, 2023 15:47 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] 🔥 ✨ Request for Feedback: Organization Discussions (Discussion #17669)




 
your welcome

**Comment by Binary-Ninja-007** [comment]
Hello, @divyaGrvgithub.

**Comment by bdon** [comment]
Hi @jayceeday - I have an org-level Discussions enabled, and am not receiving notifications for when new threads are created. I do get subscribed to notifications once I have participated in one. Is this a missing feature for org-level Discussions, or am I missing a setting?

**Reply by lars-reimann** [reply]
Try watching the `.github` repository or whichever repository you've selected to host the org-level discussions.

**Reply by jsoref** [reply]
You guessed correctly, it's https://github.com/protomaps/.github/discussions :)

**Comment by ceolinwill** [comment]
Org-level discussions are great but, as others have already mentioned, we need a way for people to subscribe to new posts on discussions. Ideally, we should have a "subscribe/unsubscribe" button that allows users to subscribe to all posts or by category.

Another feature I'd love is to have a "Discussions" tab on every repository but linking the org-level discussions. At @zoonk, I'd like to keep the conversation organized at the organization level. So, I'm not enabling "Discussions" for every repository but I'd like every repo to have the "discussions" tab linking the org-discussion page - similar to what happens with "Projects".

**Comment by marcklingen** [comment]
Just learned about org-wide discussions and really like it so far as we want to centralize the discussions across all repositories.

Slight UI glitch, whenever I am in discussions, the org settings are _active_
![CleanShot 2024-01-31 at 23 42 41@2x](https://github.com/community/community/assets/2834609/452d4435-4f30-469f-8b45-93d0602269bc)

Also, I'd prefer if org-level discussion were available on the org and repo level. Currently the main menu switches to the org level when going to discussions from our main repo. This adds additional steps to get back to the issues of our main repo: Discussions -> repos -> select repo -> go to issues

Agree with previous feedback, I'd like to make this org-level discussion available in all our other repositories within this org.

**Comment by divyaGrvgithub** [comment]
I want to express my sincere gratitude for implementing organization-level discussions. It's a step in the right direction for addressing scalability concerns within organizations managing multiple related repositories.

The approach of having a separate forum for each repository falls short when projects span multiple repositories or when there are shared aspects among them. Monitoring and moderating multiple threads per repository quickly becomes unmanageable, especially when users inadvertently post in the wrong place or fail to search for existing discussions.

However, there's a critical feature missing that limits the practicality of organization discussions: the absence of a "Discussion" tab on each repository. This omission significantly hinders users' ability to discover organization-wide discussions. Many users might assume that discussions aren't enabled for a particular repository, leading to fragmentation and missed opportunities for collaboration.

While implementing a link to a specific category within organization discussions would be ideal for improving discoverability, I understand it may pose technical challenges. Even a link to the root of organization discussions would be a significant improvement and deemed acceptable by many users.

I hope this feedback proves helpful. I'm eager to hear about any plans to address this issue, although I understand that providing specific timelines may not be feasible. Thank you for considering these suggestions as we work towards making organization-level discussions more effective and accessible.
